docs/source/reference/optimizer.md

```diff
@@ -113,6 +113,7 @@ This is the main configuration object for the optimizer.
 -   `eval_metrics: dict[str, OptimizerMetric] | None`: A dictionary of evaluation metrics to optimize. The keys are custom names for the metrics, and the values are `OptimizerMetric` objects.
 -   `numeric.enabled: bool`: Enable numeric optimization (Optuna). Defaults to `true`.
 -   `numeric.n_trials: int`: Number of numeric trials. Defaults to `20`.
+-   `numeric.sampler: str | None`: Sampling strategy for numeric optimization. `None` and `bayesian` use Optuna default: TPE for single-objective, NSGA-II for multi-objective. `grid` performs exhaustive grid search over parameter combinations. For grid search, optimizable parameters must either specify explicit `values` or provide `low`, `high`, and `step` to create the range. Defaults to `None`.
```

Comment:
Last piece of feedback (a micronit)

the type is technically a Enum -- unsure if we want to explicitly list that type here. Since its in the docs you can link to the type